pkg/compare/correlator.go

```diff
@@ -305,3 +268,58 @@ func (c *MetricsCorrelatorDecorator) addUNMatch(cr *unstructured.Unstructured) {
 	c.UnMatchedCRs = append(c.UnMatchedCRs, cr)
 	c.unMatchedLock.Unlock()
 }
+
+type FieldCorrelator struct {
+	Fields    [][]string
+	hashFunc  templateHashFunc
+	templates map[string][]*ReferenceTemplate
+}
+
+func (f *FieldCorrelator) Take(templates []*ReferenceTemplate) []*ReferenceTemplate {
```

Comment:
It's not so clear from the func name what it is supposed to do

Reply:
Fair enough. My thought process was it takes the values it needs the returns the rest. Do you have a suggestion? Perhaps ClaimTemplates?